lib/IRGen/LoadableByAddress.cpp

```diff
@@ -296,9 +296,15 @@ bool LargeSILTypeMapper::shouldTransformResults(GenericEnvironment *genEnv,
   if (!modifiableFunction(loweredTy)) {
     return false;
   }
+
   if (loweredTy->getNumResults() != 1) {
-    return false;
+    auto resultType = loweredTy->getAllResultsType();
+    auto newResultType = getNewSILType(genEnv, resultType, Mod);
+    bool hasFuncSig = containsFunctionSignature(genEnv, Mod,
+                                                resultType, newResultType);
+    return hasFuncSig;
```

Comment:
I changed the condition to compare `resultType` and `newResultType` instead of detecting function signature here.

For the second issue, it seems that we cannot simply reuse the code. Because when it comes to single-result case, `loweredTy->getAllResultsType()` would return [getType()](https://github.com/apple/swift/blob/master/lib/SIL/SILFunctionType.cpp#L69) while what we intend to have is `getSILStorageType()`. I tried doing a similar check on all results type here, and it unfortunately crashes the compiler on some test code.